src/common/http.cc

```diff
@@ -852,6 +856,25 @@ namespace Pistache::Http
             }
         }
 
+#ifdef PISTACHE_USE_CONTENT_ENCODING_GZIP
+
+        // User requested GZip compression...
+        if(contentEncoding_.has_value() &&
+           contentEncoding_.value().encoding() == Http::Header::Encoding::Gzip)
+        {
+            // Initialize constructor...
+            std::stringbuf compressorBuffer;
+            zstr::ostream compressor(&compressorBuffer);
```

Comment:
I'm personally not a fan of using an (obscure?) C++ wrapper for Zlib, especially considering that our usage is pretty basic. The `libhpptools` Debian package [has never been updated](https://tracker.debian.org/pkg/libhpptools/news/) to a new upstream version, and contains the old `zstr` version available in the <https://github.com/mateidavid/hpptools> repository, and not the more recent <https://github.com/mateidavid/zstr>.

While I could update the Debian package myself, I believe we should simply use the regular zlib functions to compress the stream, especially considering how simple it is and how little we use it.

```cpp
size_t compressed_size = compressBound(size);
std::unique_ptr compressed = std::make_unique<unsigned char>(compressed_size);
compress(compressed.get(), &compressed_size, data, size);
return putOnWire(compressed.get(), compressed_size);
```

That being said, I think there's another issue here.

By default, zlib compresses data in the `zlib` format ([RFC 1950](https://datatracker.ietf.org/doc/html/rfc1950)), while the `gzip` content encoding expects data in the `gzip` format ([RFC 1952](http://tools.ietf.org/html/rfc1952)). I don't think there is an easy one-way zlib function like `compress()` that produces `gzip`-formatted data, but we could simply choose to handle the `deflate` content encoding instead, which quite conveniently expects `zlib`-formatted data.